python/cudf/cudf/core/column/categorical.py

```diff
@@ -1201,6 +1210,20 @@ def find_and_replace(
             ordered=self.dtype.ordered,
         )
 
+    def isnull(self) -> ColumnBase:
+        """Identify missing values in a Column.
+        """
+        result = libcudf.unary.is_null(self)
+
+        if self.categories.dtype.kind == "f":
+            # Need to consider `np.nan` values incase
+            # of a float column
+            result = result | libcudf.unary.is_nan(
+                self.astype(self.categories.dtype)
+            )
```

Comment:
Instead of decoding the column, I'm wondering if it would be more efficient to check if `NaN` is in the categories, if it is, get the associated code, and then generate the boolean mask from it and then run a `logical_or` operation with `result`.

Reply:
Yup, updated code to perform the same. This approach would be much efficient than decoding the column